craterpy/classes.py

```diff
@@ -469,3 +475,186 @@ def plot(self, name="", ax=None, alpha=0.2, **kwargs):
+        if drop_private:
+            private_cols = [col for col in data.columns if col.startswith("_") and col != data.geometry.name]
+            data = data.drop(columns=private_cols)
+        
+        # Filter properties if specified
+        if properties is not None:
+            # Make sure the geometry column is included
+            geometry_name = data.geometry.name
+            keep_cols = list(set(properties + [geometry_name]))
+            missing_cols = [col for col in keep_cols if col not in data.columns]
+            if missing_cols:
+                raise ValueError(f"Properties not found: {missing_cols}")
+            data = data[keep_cols]
+        
+        # Convert to the target CRS if specified
+        if crs is not None:
+            # Handle various input types for CRS
+            try:
+                target_crs = CRS.from_user_input(crs)
+                if data.crs != target_crs:
+                    data = data.to_crs(target_crs)
+            except Exception as e:
+                raise ValueError(f"Error converting to CRS '{crs}': {str(e)}")
+        
+        # Export to GeoJSON
+        if filename is not None:
+            data.to_file(filename, driver="GeoJSON")
+            return None
+        else:
+            # Return as string
+            return data.to_json()
+
+
+    @classmethod
+    def read_shapefile(cls, filename, body="Moon", units="m"):
+        """
+        Read crater data from a shapefile or GeoJSON file.
+        
+        Parameters
+        ----------
+        filename : str
+            Path to the shapefile or GeoJSON file.
+        body : str, optional
+            Planetary body, e.g. Moon, Vesta (default: Moon).
+            If the file contains a 'body' or 'planet' field, that value will be used instead.
+        units : str, optional
+            Length units of radius/diameter, m or km (default: m).
+            If the file contains a 'units' field, that value will be used instead.
+        
+        Returns
+        -------
+        CraterDatabase
+            A new CraterDatabase instance containing the data from the file.
+            
+        Notes
+        -----
+        This method assumes the file was previously created by CraterDatabase.to_geojson()
+        or has a compatible format with lat/lon coordinates and radius or diameter information.
+        
+        If the file contains different coordinate column names than expected, this method
+        will attempt to identify them by common names (e.g., 'lat', 'latitude', 'lon', 'longitude').
+        """
+        gdf = gpd.read_file(filename)
+        
+        # Check if the file has body or units info
+        if 'body' in gdf.columns and gdf['body'].nunique() == 1:
+            body = gdf['body'].iloc[0]
+        elif 'planet' in gdf.columns and gdf['planet'].nunique() == 1:
+            body = gdf['planet'].iloc[0]
+            
+        if 'units' in gdf.columns and gdf['units'].nunique() == 1:
```

Comment:
similarly i don't think we ever write a units column to the exported geojson. we could write units as metadata in to_geojson() and try to read it from there automatically. might need an extra metadata read step (which may be useful to guarantee one-to-one reads and writes): https://stackoverflow.com/questions/73023509/how-to-get-value-from-a-geojson-that-was-converted-into-a-geopandas-geodatafram